go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -144,28 +157,44 @@ func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase
 	return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", resource.State)
 }
 
-func getFinalEndpoint(taskType, defaultEndpoint string, endpointForTaskTypes map[string]string) string {
+func getFinalEndpoint(taskType string, defaultEndpoint GrpcEndpoint, endpointForTaskTypes map[string]GrpcEndpoint) GrpcEndpoint {
 	if t, exists := endpointForTaskTypes[taskType]; exists {
 		return t
 	}
 
 	return defaultEndpoint
 }
 
-func getClientFunc(ctx context.Context, endpoint string, connectionCache map[string]*grpc.ClientConn) (service.AsyncAgentServiceClient, error) {
-	conn, ok := connectionCache[endpoint]
+func getClientFunc(ctx context.Context, endpoint GrpcEndpoint, connectionCache map[string]*grpc.ClientConn) (service.AsyncAgentServiceClient, error) {
+	conn, ok := connectionCache[endpoint.Endpoint]
 	if ok {
 		return service.NewAsyncAgentServiceClient(conn), nil
 	}
+
 	var opts []grpc.DialOption
-	var err error
 
-	opts = append(opts, grpc.WithInsecure())
-	conn, err = grpc.Dial(endpoint, opts...)
+	if endpoint.Insecure {
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	} else {
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+
+		creds := credentials.NewClientTLSFromCert(pool, "")
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	}
+
+	if endpoint.DefaultServiceConfig != "" {
```

Comment:
I think `len(endpoint.DefaultServiceConfig) > 0` is the idiomatic golang way to do this.

Reply:
Sure. I just followed existing code I found.